packages/world/src/systems/OreSystem.sol

```diff
@@ -4,87 +4,84 @@ pragma solidity >=0.8.24;
 
+import { InventoryObjects } from "../codegen/tables/InventoryObjects.sol";
+import { TotalMinedOreCount } from "../codegen/tables/TotalMinedOreCount.sol";
+import { MinedOrePosition, MinedOrePositionData } from "../codegen/tables/MinedOrePosition.sol";
 import { ObjectType } from "../codegen/tables/ObjectType.sol";
 import { Player } from "../codegen/tables/Player.sol";
 import { Position } from "../codegen/tables/Position.sol";
 import { PlayerStatus } from "../codegen/tables/PlayerStatus.sol";
 import { LastKnownPosition } from "../codegen/tables/LastKnownPosition.sol";
 import { ReversePosition } from "../codegen/tables/ReversePosition.sol";
 import { ActionType } from "../codegen/common.sol";
-import { TerrainCommitment, TerrainCommitmentData } from "../codegen/tables/TerrainCommitment.sol";
-import { Commitment } from "../codegen/tables/Commitment.sol";
-import { BlockPrevrandao } from "../codegen/tables/BlockPrevrandao.sol";
+import { OreCommitment } from "../codegen/tables/OreCommitment.sol";
 
-import { AirObjectID, PlayerObjectID, AnyOreObjectID, LavaObjectID, CoalOreObjectID } from "../ObjectTypeIds.sol";
-import { inWorldBorder, getRandomNumberBetween0And99 } from "../Utils.sol";
+import { AirObjectID, PlayerObjectID, LavaObjectID, CoalOreObjectID } from "../ObjectTypeIds.sol";
+import { inWorldBorder } from "../Utils.sol";
 import { requireValidPlayer, requireInPlayerInfluence } from "../utils/PlayerUtils.sol";
 import { notify, InitiateOreRevealNotifData, RevealOreNotifData } from "../utils/NotifUtils.sol";
 import { TerrainLib } from "./libraries/TerrainLib.sol";
 import { ObjectTypeId } from "../ObjectTypeIds.sol";
 import { EntityId } from "../EntityId.sol";
+import { ChunkCoord } from "../Types.sol";
+import { COMMIT_EXPIRY_BLOCKS, COMMIT_HALF_WIDTH } from "../Constants.sol";
+
+// TODO: copied from voxel coords. Figure out way to unify coordinate utils
+function inSurroundingCube(
+  ChunkCoord memory cubeCenter,
+  int32 halfWidth,
+  ChunkCoord memory checkCoord
+) pure returns (bool) {
+  bool isInX = checkCoord.x >= cubeCenter.x - halfWidth && checkCoord.x <= cubeCenter.x + halfWidth;
+  bool isInY = checkCoord.y >= cubeCenter.y - halfWidth && checkCoord.y <= cubeCenter.y + halfWidth;
+  bool isInZ = checkCoord.z >= cubeCenter.z - halfWidth && checkCoord.z <= cubeCenter.z + halfWidth;
+
+  return isInX && isInY && isInZ;
+}
 
 contract OreSystem is System {
   using VoxelCoordLib for *;
 
-  function initiateOreReveal(VoxelCoord memory coord) public {
-    require(inWorldBorder(coord), "Cannot reveal ore outside world border");
-
-    (EntityId playerEntityId, VoxelCoord memory playerCoord, ) = requireValidPlayer(_msgSender());
-    requireInPlayerInfluence(playerCoord, coord);
-    if (TerrainCommitment._getBlockNumber(coord.x, coord.y, coord.z) != 0) {
-      revealOre(coord);
-      return;
-    }
+  function oreChunkCommit(ChunkCoord memory chunkCoord) public {
```

Comment:
why are `OreCommitments` per chunk coord vs per the ore coord?

Reply:
We could do it per ore, but I think it makes sense making it per chunk while exploring, having it per ore doesn't bring any benefits imo. I just found an issue though, I'm only using the block randomness when mining, not the block coord